test/val/val_atomics_test.cpp

```diff
@@ -488,7 +515,7 @@ OpAtomicStore %f32vec4_var %device %relaxed %f32_1
 
 TEST_F(ValidateAtomics, AtomicStoreWrongPointerStorageType) {
   const std::string body = R"(
-%f32_var_function = OpVariable %f32_ptr_function Function
+%f32_var_function = OpVariable %f32_ptr_uniformconstant Function
```

Comment:
Why did this change? The error message is unchanged and this OpVariable is incorrect in terms of storage classes now.

Reply:
I made this change because an atomic store on a Function pointer is valid for at least some environments now, whereas an atomic store on a UniformConstant pointer is invalid for all environments.  I'm totally open to changing this back though.

Since this test compiles for the "universal" environment it will "fail correctly" even with a Function pointer.  Is that preferred?

If I change this back to a Function pointer I think I should also remove the new AtomicStoreVulkanFunctionPointerStorageType test, since it's effectively already covered here?

Reply:
I'm ok with changing this to test the UniformConstant storage class, but I'd prefer the rest of the assembly is still valid. I don't like relying on the order of checks in the validator so I just want the mismatch in storage class between pointer and variable fixed.

Reply:
Ah, I see now.  I'll fix this and the formatting issues.

Reply:
I believe both this and the formatting issues are fixed.